Assets/RGLUnityPlugin/Scripts/SceneManager.cs

```diff
@@ -409,5 +410,41 @@ private static bool IsNotActiveOrParentHasLidar(GameObject gameObject)
         {
             return !gameObject.activeInHierarchy || gameObject.GetComponentsInParent<LidarSensor>().Length != 0;
         }
+
+        private static bool TryCreateRGLObject<T>(T meshSource, out IRGLObject rglObject) where T : UnityEngine.Object
```

Comment:
To consider:
maybe move this switch-like a̶b̶o̶m̶i̶n̶a̶t̶i̶o̶n̶  code into the templated rglObject constructor?

Reply:
or switch at least

Reply:
Moved to `RGLObjectHelper`. Calling the RGLObject static method needs generic type specialization which is unnecessary in this case.